docs/development.md

```diff
@@ -67,93 +106,147 @@ docker compose down -v
-```
-
-You can connect to the instance manually
-
-```bash
-docker run -d -p 3000:3000 --network nmdc-server_default mongoclient/mongoclient
-```
-
-In order to populate the database, you must create a `.env` file in the top
-level directory containing mongo credentials.
-
-```bash
-# .env
-NMDC_MONGO_USER=changeme
-NMDC_MONGO_PASSWORD=changeme
-```
-
-With that file in place, populate the docker volume by running,
-
-```bash
-docker-compose run backend nmdc-server truncate # if necessary
-docker-compose run backend nmdc-server migrate
-docker-compose run backend nmdc-server ingest -vv --function-limit 100
-```
-
-# Running the client
-
-Run the client in development mode.
-
-``` bash
-cd web/
-yarn
+cd web
 yarn serve
 ```
 
-View main application at `http://127.0.0.1:8081`
+View the main application at `http://127.0.0.1:8081/`. Changes to files in the `web` directory will automatically trigger a reload in your browser. 
+
+> **Note**: the frontend application will still be served via Docker Compose on port `8080`, but it will not pick up changes to the `web` directory automatically. Be aware of which port you are accessing when doing frontend development.
 
-## Why not `localhost`?
+### Why not `localhost`?
 
 It is recommended to use `127.0.0.1` instead of `localhost` for local development. This is because `localhost` is **not** allowed as a redirect URI for an ORCID client. The workaround is to register `127.0.0.1` as a redirect URI with ORCID and to use subsequently visit `127.0.0.1` for local testing.
 
-# Testing
+## Running ingest
+
+> **Note**: This is not generally required unless you are specifically working on the ingest code. If you are working on the web application, simply [loading from a recent production backup](#load-production-data) is sufficient.
+
+1. Ensure that you have completed the sections above of configuring [NERSC credentials](#nersc-credentials) and [MongoDB credentials](#mongodb-credentials). 
+2. Obtain a new SSH key from NERSC if you haven't done so in the last 24 hours.
+    ```bash
+    sshproxy.sh -u $NERSC_USER
+    ```
+3. Set up an active SSH tunnel to the dev and production MongoDB instances if you do not already have one established.
+    ```bash
+    ssh \
+      -L 37018:mongo-loadbalancer.nmdc-dev.production.svc.spin.nersc.org:27017 \
+      -L 37019:mongo-loadbalancer.nmdc.production.svc.spin.nersc.org:27017 \
+      -o ServerAliveInterval=60 \
+      -f \
+      -N \
+      -l $NERSC_USER \
+      -i ~/.ssh/nersc \
+      dtn01.nersc.gov
+    ```
+    > That command will set up SSH port forwarding such that your computer can access the dev MongoDB server at `localhost:37018` and the prod MongoDB server at `localhost:37019`. 
+  
+    > From within a Docker container `host.docker.internal` can be used to access the `localhost` of your computer. When ingesting from the dev or prod MongoDB instances, be sure to set `NMDC_MONGO_HOST=host.docker.internal` in your `.env` file.
```

Comment:
```suggestion
    > From within a Docker container, `host.docker.internal` can be used to access the `localhost` of your computer. When ingesting from the dev or prod MongoDB instances, be sure to set `NMDC_MONGO_HOST=host.docker.internal` in your `.env` file.
```

If I remember correctly, `host.docker.internal` is not the special hostname that Docker environments running on Linux hosts use. I will check my notes and report back.

Reply:
Correction: `host.docker.internal` _can_ be used in a Docker container running on a Linux host, provided the container has been configured appropriately.

Here's an example of where someone configured a container appropriately:

https://github.com/microbiomedata/nmdc-edge/blob/1c6bc57a6e9969b18c6a4c03a65417809b13a284/docs/docker-compose.prod.yml#L58-L61

I think some of the Kitware team members will be familiar with the situation as I think they use Linux machines for development.

Reply:
Oh that's good to know. It's something that I've always used, but then again I've always worked on a Mac. I am just now realizing that the [documentation](https://docs.docker.com/desktop/features/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host) I referenced in the original PR description is in the Docker Desktop section. So maybe it's not OS-dependent, but rather whether or not you're using Docker Desktop?

Reply:
I develop on linux (Ubuntu 22.04) and have struggled with `host.docker.internal` in the past (although I never knew about how to configure it "correctly," so grain of salt) (thanks @eecavanna)

Previously I used the following command to connect to mongo:
```
docker run --rm -it -p 27017:27017 --network nmdc-server_default --name tunnel kroniak/ssh-client ssh -o StrictHostKeyChecking=no -L 0.0.0.0:27017:mongo-loadbalancer.nmdc.production.svc.spin.nersc.org:27017 $NERSC_USER@dtn01.nersc.gov '/bin/bash -c "while [[ 1 ]]; do echo heartbeat; sleep 300; done"'
```

I can test using `host.docker.internal` combined with `extra_hosts` to see if there's an issue with that.

Reply:
I'm struggling to get `host.docker.internal` working for me. I'm curious if @marySalvi would have better luck, or has used it before. I've tried messing around with my docker daemon config (using ChatGPT as a guide, so who knows) to no avail

Reply:
@naglepuff This might be a long shot, but which version of docker are you using, `docker --version`?

Reply:
Hi @naglepuff, I want to clarify this statement:

> I'm struggling to get `host.docker.internal` working for me.

Do you mean that, even when the a service definition in a `docker-compose.yml` file has...

```yaml
services:
  my-svc:
    # ...

    extra_hosts:
      - "host.docker.internal:host-gateway"
```

...in it, and you have some web server on your Linux laptop listening on port 8000, like this...

```console
# Start a web server
$ python -m http.server
Serving HTTP on :: port 8000 (http://[::]:8000/) ...
```

..., a container running the service (e.g. `my-svc`) is still unable to access that web server via `http://host.docker.internal:8000`?

I assumed that would work, based on what I observed when trying to set up a containerized NMDC EDGE stack on a Linux VM a few months ago.